### PR TITLE
support generate test data with seed data input

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary
support generate test data with seed data input

https://github.com/finos/legend-studio/assets/73408381/b393b54f-3700-4257-94e1-2af7c51274fb

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts [hunk 1]
```diff
@@ -125,6 +137,88 @@ const createMockPrimitiveValueSpecification = (
   );
 };
 
+export const relationalDataTypeToPrimitiveType = (
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts:140]: lets avoid duplicate cod!!
`getPrimitiveTypeFromRelationalType``

File: packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts [hunk 2]
```diff
@@ -125,6 +137,88 @@ const createMockPrimitiveValueSpecification = (
...
+    relationalDataType instanceof RelationalDateTypeFloat ||
+    relationalDataType instanceof RelationalDateTypeDouble ||
+    relationalDataType instanceof RelationalDateTypeInteger
+  ) {
+    return PrimitiveType.NUMBER;
+  }
+  return PrimitiveType.STRING;
+};
+
+export const createMockPrimitiveValueSpecificationFromRelationalDataType = (
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts:170]: is this function needed. I dont really want relational things in this file

File: packages/legend-graph/src/graph-manager/protocol/pure/v1/model/packageableElements/store/relational/model/V1_TablePtr.ts [hunk 3]
```diff
@@ -33,3 +34,12 @@ export class V1_TablePtr implements Hashable {
     ]);
   }
 }
+
+export const V1_buildTablePtr = (protocol: V1_TablePtr): TablePtr => {
```
Comment (MauricioUyaguari) [inline packages/legend-graph/src/graph-manager/protocol/pure/v1/model/packageableElements/store/relational/model/V1_TablePtr.ts:38]: lets keep the protocol models clean. we should have this in the relational helper builder

File: packages/legend-graph/src/graph-manager/protocol/pure/v1/transformation/pureGraph/from/V1_DatabaseTransformer.ts [hunk 4]
```diff
@@ -168,6 +169,17 @@ const transformRelationalDataType = (
   );
 };
 
+export const V1_transformTablePointer = (ptr: TablePtr): V1_TablePtr => {
+  const tablePtr = new V1_TablePtr();
+  tablePtr.database = ptr.database;
+  // NOTE: Sometimes, we interpret this, so to maintain roundtrip stability, we need to handle this differrently
+  // See https://github.com/finos/legend-studio/issues/295
```
Comment (MauricioUyaguari) [inline packages/legend-graph/src/graph-manager/protocol/pure/v1/transformation/pureGraph/from/V1_DatabaseTransformer.ts:176]: is this comment relevant for this function? or did we copy it from below?

File: packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/model/TablePtr.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { hashArray, type Hashable } from '@finos/legend-shared';
+import { CORE_HASH_STRUCTURE } from '../../../../../../Core_HashUtils.js';
+
+export class TablePtr implements Hashable {
```
Comment (MauricioUyaguari) [inline packages/legend-graph/src/graph/metamodel/pure/packageableElements/store/relational/model/TablePtr.ts:20]: lets move this to the service folder as its not really a metamodel class

File: packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestDataEditor.tsx [hunk 6]
```diff
@@ -201,6 +455,15 @@ export const ConnectionTestDataEditor = observer(
       }
     };
 
+    const generateTestDataWithSeedData = (): void => {
+      connectionTestDataState.setUseSeedDataInputModal(true);
+      connectionTestDataState.setNewTableIdentifierState([]);
+      const tables = connectionTestDataState.getAvailableTables();
+      if (tables[0]) {
+        connectionTestDataState.addNewTableIdentifierState(tables[0]);
+      }
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestDataEditor.tsx:464]:
```suggestion
      const tables = connectionTestDataState.getAvailableTables();
      if (tables[0]) {
        connectionTestDataState.addNewTableIdentifierState(tables[0]);
      }
```
      const table = getNullableFirstEntry(connectionTestDataState.getAvailableTables());
      if (table) {
        connectionTestDataState.addNewTableIdentifierState(table);
      }

File: packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts (hunk 1 above)
Comment (YannanGao-gs) [reply packages/legend-query-builder/src/stores/shared/ValueSpecificationEditorHelper.ts:140]: i didn't realize that this function exists..

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**